File: gestion-materiel-frontend/src/lib/api/public-scan.ts
/**
 * Public (unauthenticated) API for employee self-service scan.
 * Uses a separate axios instance — NO JWT, NO 401 redirect.
 */
import axios from "axios";
import { config } from "@/lib/config";

const publicHttp = axios.create({ baseURL: config.apiBaseUrl });

// ---------- Types ----------

export interface PublicAssetOut {
  public_id: string;
  name: string;
  category: string;
  status: string;
  plate: string | null;
  km_current: number | null;
  epi_type: string | null;
}

export interface PublicActionResult {
  success: boolean;
  message: string;
  asset: PublicAssetOut;
}

// ---------- API calls ----------

export async function publicScanAsset(publicId: string): Promise<PublicAssetOut> {
  const { data } = await publicHttp.get<PublicAssetOut>(`/public/scan/${publicId}`);
  return data;
}

export async function publicTakeAsset(payload: {
  public_id: string;
  employee_code: string;
  km_value?: number | null;
  notes?: string | null;
}): Promise<PublicActionResult> {
  const { data } = await publicHttp.post<PublicActionResult>("/public/scan/take", payload);
  return data;
}

export async function publicReturnAsset(payload: {
  public_id: string;
  employee_code: string;
  km_value?: number | null;
  notes?: string | null;
}): Promise<PublicActionResult> {
  const { data } = await publicHttp.post<PublicActionResult>("/public/scan/return", payload);
  return data;
}

File: backend/app/schemas/public_scan.py
from pydantic import BaseModel


class PublicAssetOut(BaseModel):
    """Limited asset info for employee self-service display."""

    public_id: str
    name: str
    category: str  # VEHICLE / EPI
    status: str  # AVAILABLE / ASSIGNED / MAINTENANCE / RETIRED

    # Vehicle
    plate: str | None = None
    km_current: int | None = None

    # EPI
    epi_type: str | None = None

    model_config = {"from_attributes": True}


class PhotoOut(BaseModel):
    id: int
    category: str  # STATE / DAMAGE
    url: str


class PublicActionResult(BaseModel):
    success: bool
    message: str
    asset: PublicAssetOut
    photos: list[PhotoOut] = []
